### DISCUSSION TITLE
SkiaSharp view for Godot Game Engine

### DISCUSSION THREAD

**Discussion by Ilnazz** [mono/SkiaSharp#2656] [Show and tell]
Hello. Thanks for the library.
I created a SkiaSharp view for the Godot Game Engine.
Link to the repository:
https://github.com/Ilnazz/SkiaSharp.Views.Godot

What do you think about this?
Is it possible to include this view to your library?

**Comment by Ilnazz** [comment]
Good time of day. Please look at my previous message, maybe you missed it.

**Reply by taublast** [reply]
Looks very interesting, do you happen to have a use case, any sample?

**Reply by Ilnazz** [reply]
Good time of day. Thank you for your attention to the question.
An example of using this SkiaSharp view is the my port of the LiveCharts2 library for Godot Game Engine - [LiveCharts2 plugin for Godot](https://github.com/Ilnazz/godot-live-charts).
It is used there in a single place - the Motion Canvas class is inherited from the SKControl - [MotionCanvas](https://github.com/Ilnazz/LiveChartsCore.SkiaSharpView.Godot/blob/master/src/skiasharp/LiveChartsCore.SkiaSharpView.Godot/MotionCanvas.cs).

**Comment by mattleibow** [comment]
This looks interesting... Will you be willing to make a PR?

You can just add the code and maybe a basic sample that is runnable for testing.

Does this same code run on all platforms? I see you reference desktop, which is limited.

**Reply by Ilnazz** [reply]
Good time of day. I added a sample project and tested it on Windows and Linux Ubuntu only. I hurried with the proposal to include this View to the SkiaSharp repository, since it is necessary to implement the missing extension methods and test this View on other platforms. At the moment, I don't have time to do this, so it's worth putting this matter on hold.
